pelicun/tools/DL_calculation.py

```diff
@@ -1043,47 +1253,63 @@ def _demand_assessment(
     assessment.demand.load_sample(convert_to_SimpleIndex(demand_sample, axis=1))
 
 
-def _asset_definition(config, assessment, color_codes):
-
-    # if no asset data are specified, simply return
-    if is_unspecified(config, 'DL/Asset'):
-        return
+def _asset_definition(
+    assessment,
+    num_stories,
+    component_assignment_file,
+    collapse_fragility_demand_type,
+    add_irreparable_damage_columns,
+    component_sample_file,
+    color_codes,
+):
 
     # retrieve the demand sample
     demand_sample = assessment.demand.save_sample()
 
     # set the number of stories
-    if get(config, 'DL/Asset/NumberOfStories', default=False):
-        assessment.stories = int(get(config, 'DL/Asset/NumberOfStories'))
+    if num_stories:
+        assessment.stories = num_stories
+
+    # We either accept a `component_assignment_file` or a
+    # `component_sample_file`, not both.
+    if component_assignment_file is not None and component_sample_file is not None:
+        raise ValueError(
+            'Both `component_assignment_file` and '
+            '`component_sample_file` are provided. '
+            'Please provide only one.'
+        )
```

Comment:
The intention was to gradually move this code in `assessment.py` in the form of `Assessment` methods which can be used without `DL_calculation.py`.

Reply:
Yes, but you would still need to parse the config file and validate the inputs. I thought that such checks belong there rather than in the actual calculation process regardless of that process being in assessment.py or DL_calculation.py.